client/src/modules/SearchBar/Components/SearchBar.tsx

```diff
@@ -37,50 +37,33 @@ export const SearchBar = ({
     if (query.toLowerCase() !== '') {
       setTimeout(() => {
         axios
-          .post(`/api/getClassesByQuery`, { query: query })
+          .post(`/api/getResultsFromQuery`, { query: query })
           .then((response) => {
-            const queryCourseList = response.data.result
-            if (queryCourseList.length !== 0) {
-              setCourses(queryCourseList)
-            } else {
-              setCourses([])
-            }
-          })
-          .catch((e) => {
-            setCourses([])
-            console.log('Getting courses failed!')
-          })
+            const subjectList = response.data.result.subjects
```

Comment:
Will also say - _ideally_, we want the our endpoint to return only the data we want to work with. 

Further down the code, we slice each of these lists, ie. (0,3) and (0,5). The endpoint should only give us 3 items in subjects/professors and 5 in courses I think? 

IE. if we search 'cs' => I checked how big the course list length is (200) which is 195 data points we don't need the frontend to ever be exposed to => longer loading times and slower rendering (although these objects aren't that big + shouldn't make a big diff , it's still best practice.)